.github/actions/get_unpublished_tags/action.yml

```diff
@@ -0,0 +1,39 @@
+name: 'Fetch unpublished tags'
+description: 'This actions fetches all unpublished tags'
+inputs:
+  indexer-repo-path:
+    description: 'The path to the indexer repo'
+    required: true
+  version:
+    description: 'autoagora-indexer-service version'
+    required: false
+outputs:
+  tags:
+    description: 'The unpublished tags'
+    value: ${{ steps.get_diff.outputs.tags }}
+runs:
+  using: "composite"
+  steps:
+    - name: Get the latest package versions
+      id: fetch-package-versions
+      run: |
+        echo "versions=$(skopeo list-tags docker://ghcr.io/nut-a/autoagora-indexer-service | jq -r -c '.Tags')" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Get the latest indexer versions
+      id: get-latest-indexer-versions
+      run: |
+        cd ${{ inputs.indexer-repo-path }}
+        echo "versions=$(git tag --sort=-creatordate --format='%(refname:short)' | awk 'BEGIN { printf "[" } { if (NR<=5) { printf "\"%s\"", $0; if (NR != 5) printf "," } } END { printf "]" }')" >> $GITHUB_OUTPUT
```

Comment:
I'd get the indexer repo's container images tags rather than git tags, since the `autoagora-indexer-srevice` container build really only cares about the indexer's container images: https://github.com/semiotic-ai/autoagora-indexer-service/blob/ab2c3f1e161f0161daab46aa61fe768595d712ca/Dockerfile#L11

You could use this for example:
```sh
skopeo list-tags docker://ghcr.io/graphprotocol/indexer-service | jq -r -c '.Tags | map(select(test("^v\\d+\\.\\d+\\.\\d+$")))'
```

Reply:
Make sense